### ISSUE TITLE
Accessibility: make default palette colorblind friendly?

### ISSUE THREAD

**Issue by tpoisot** [JuliaPlots/Plots.jl#1144]
I think it would be a good thing to have a default palette that works for everyone -- the current palette is not good for this (after running some plots through a simulator).

I would suggest to replace the default by: 

![palette](http://mkweb.bcgsc.ca/colorblind/img/colorblindness.palettes.trivial.png)

In fact, if you tell me where to make the change, I'll be glad to submit a PR.

**Comment by mkborregaard** [comment]
I'm up for discussing this, I've thought about it for some time. But let's have a good full discussion of it. It'll break everyone's plots, not everyone's going to like it, and the current one is based on `distinguishable_colors` which is an ecosystem-wide standard. In fact it would make sense to open an issue on Colors.jl with this suggestion.

**Comment by tpoisot** [comment]
I think it would "break" everyone's plot for the better (we submitted a paper discussing, in part, why graphic packages should be colorblind friendly by default), but point taken.

The idea of distinguishable colors makes sense *but* they're only distinguishable if you can see the full spectrum. So this comes with a huge accessibility issue built-in.

I'm of course happy to open an issue on Colors.jl -- just out of curiosity, where's the code in Plots.jl responsible for setting the default colors?

**Comment by daschw** [comment]
I think they are set by the defaults here: https://github.com/JuliaPlots/PlotUtils.jl/blob/master/src/color_utils.jl#L46

**Comment by mkborregaard** [comment]
>  think it would "break" everyone's plot for the better

agreed

**Comment by daschw** [comment]
A first step could be to submit create a new colorblind friendly theme in https://github.com/JuliaPlots/PlotThemes.jl

**Comment by mkborregaard** [comment]
Looking at it we could actually perhaps just change `color_bases` without involving Colors.jl. That's worth considering. I think that's set by `palette`?
So, I agree with @daschw - let's start by defining a number of color-blind friendly palettes (perhaps in PlotThemes, that's not a bad idea, but we want to be able to change it without changing the other color options, so could be PlotUtils instead). Then we can play around and experiment for a while before committing to a new default theme.

**Comment by tpoisot** [comment]
I'll need a crash course in the Plots ecosystem... So if I get it right, it's possible to create a new palette in PlotUtils? And then this can be called in Plots?

Sorry for the very basic questions, but I'm trying to figure out how things are organized.

**Comment by mkborregaard** [comment]
Hm it's bit involved in fact. You can pass in a name of a color palette, and Plots will then take values from that color palette so that each new value is as far from the others as possible on that gradient (here: https://github.com/JuliaPlots/PlotUtils.jl/blob/master/src/color_utils.jl#L69) . If instead you don't specify a gradient, it'll use `distinguishable colors`, based on the background color and the short color gradient ´[colorant"steelblue",colorant"orangered"]´ here https://github.com/JuliaPlots/PlotUtils.jl/blob/master/src/color_utils.jl#L47 . 
The resulting color gradient looks like this:
<img width="485" alt="skaermbillede 2017-10-08 kl 00 05 55" src="https://user-images.githubusercontent.com/8429802/31312185-927eec92-abbc-11e7-8a64-87d386d5b2c9.png">

It will then sample from this color gradient, again to space each new series as much as possible. 
I think the right approach would be to change the ´[colorant"steelblue",colorant"orangered"]´ part to change the basic colors and still let the line 47 function deal with it.

Until that, to try out things, it would make sense simply to add some more color palettes, perhaps in a new color library called `:discrete` or something. Adding that would just be making a new file under gradients in PlotUtils and following the protocol from the others,

e.g.
```
register_color_library(:discrete, ColorLibrary())
register_gradient_colors(:awesome_palette, [:color1, :color2], :discrete)
``` 
 Then we can just call
```
gr(palette = :awesome_palette)
```
(or put a default in the .juliarc.jl file) and experiment with how it works in practice.

**Comment by mkborregaard** [comment]
If you only want to add one or two palettes (like the one you suggest above) you could also just drop it in the :misc library for now ( https://github.com/JuliaPlots/PlotUtils.jl/blob/master/src/color_gradients.jl#L69 ) - then it's just one or two lines of code.

**Comment by mkborregaard** [comment]
Hey @tpoisot check this PR just merged by @daschw (you can just read the first and last posts), if you check out master of Plots and PlotThemes you can use your suggested colorscheme from now on. Let's get some experience with using it!

**Comment by mkborregaard** [comment]
I've used this theme for the last 2 months, and I'm not super convinced.

**Comment by daschw** [comment]
Yes, the yellow often seems a little bit too bright to have enough contrast to the white bg for me.

**Comment by tpoisot** [comment]
I get your point, but it's not about *aesthetics*. It's about *accessibility*.

**Comment by mkborregaard** [comment]
It's about both to me. Especially if we're choosing a new default color scheme. This is by no means the only color-blind friendly color scheme out there.

**Comment by tpoisot** [comment]
I agree in principle - but I think if we have to chose, accessibility should be a priority.

**Comment by mkborregaard** [comment]
In my view both argue against the current default color scheme, at least...

**Comment by mkborregaard** [comment]
@tpoisot you'll be happy to see that the color scheme you're suggesting here is currently the default for StatsMakie: https://nextjournal.com/piever/statsmakie
